server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -1186,6 +1227,431 @@ protected void pushToUpdateQueue(final List<NetworkVO> networks) throws Interrup
+        } else {
+            updateRouterConnectivityHealthCheck(routerId, true, "Successfully fetched data");
+            parseAndMergeDbHealthChecks(routerId, answer.getMonitoringResults());
+        }
+
+        return true;
+    }
+
+    protected class UpdateRouterHealthChecksConfigDataTask extends ManagedContextRunnable {
+        public UpdateRouterHealthChecksConfigDataTask() {
+        }
+
+        @Override
+        protected void runInContext() {
+            try {
+                final List<DomainRouterVO> routers = _routerDao.listByStateAndManagementServer(VirtualMachine.State.Running, mgmtSrvrId);
+                s_logger.debug("Found " + routers.size() + " running routers. ");
+
+                for (final DomainRouterVO router : routers) {
+                    updateRouterHealthCheckData(router);
+                }
+            } catch (final Exception ex) {
+                s_logger.error("Fail to complete the UpdateRouterHealthChecksConfigDataTask! ", ex);
+            }
+        }
+    }
+
+    private SetMonitorServiceCommand createMonitorServiceCommand(DomainRouterVO router, List<MonitorServiceTO> services,
+                                                                 boolean reconfigure, boolean deleteFromProcessedCache) {
+        final SetMonitorServiceCommand command = new SetMonitorServiceCommand(services);
+        command.setAccessDetail(NetworkElementCommand.ROUTER_IP, getRouterControlIP(router));
+        command.setAccessDetail(NetworkElementCommand.ROUTER_NAME, router.getInstanceName());
+        command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_ENABLED, RouterHealthChecksEnabled.valueIn(router.getDataCenterId()).toString());
+        command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_BASIC_INTERVAL, RouterHealthChecksBasicInterval.value().toString());
+        command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_ADVANCED_INTERVAL, RouterHealthChecksAdvancedInterval.value().toString());
+        command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_EXCLUDED, RouterHealthChecksToExclude.valueIn(router.getDataCenterId()));
+        command.setAdditionalData(getAdditionalDataForRouterHealthChecks(router));
+        command.setReconfigureAfterUpdate(reconfigure);
+        command.setDeleteFromProcessedCache(deleteFromProcessedCache); // As part of updating
+        return command;
+    }
+
+    private boolean updateRouterHealthCheckData(DomainRouterVO router) {
+        if (!RouterHealthChecksEnabled.valueIn(router.getDataCenterId())) {
+            return false;
+        }
+
+        SetMonitorServiceCommand command = createMonitorServiceCommand(router, null,true, true);
+        String controlIP = getRouterControlIP(router);
+        if (StringUtils.isBlank(controlIP) || controlIP.equals("0.0.0.0")) {
+            s_logger.warn("Skipping update data on router " + router.getUuid() + " because controlIp is not correct.");
```

Comment:
if the control ip is not correct, this would either be an error because it should be or informational, because it is a permitted usecase. warning does not make sense, does it?

Reply:
good point. I've moved to debug.